avm/res/data-factory/factory/main.bicep

```diff
@@ -235,11 +235,9 @@ module dataFactory_integrationRuntimes 'integration-runtime/main.bicep' = [
       dataFactoryName: dataFactory.name
       name: integrationRuntime.name
       type: integrationRuntime.type
-      integrationRuntimeCustomDescription: integrationRuntime.?integrationRuntimeCustomDescription ?? 'Managed Integration Runtime created by avm-res-datafactory-factories'
-      managedVirtualNetworkName: contains(integrationRuntime, 'managedVirtualNetworkName')
-        ? integrationRuntime.managedVirtualNetworkName
-        : ''
-      typeProperties: contains(integrationRuntime, 'typeProperties') ? integrationRuntime.typeProperties : {}
+      integrationRuntimeCustomDescription: integrationRuntime.?integrationRuntimeCustomDescription
```

Comment:
With these changes here, have you verified that when using the MAX bicep file that the ADF that is created has two IR's and a few linked services like SQL and Storage account as that is some of the changes I made previously. As long as functionality is not taken away with these proposed changes then its fine. But we must be able to add many IR's and many linked services

Reply:
It should continue to work as before without a problem. But let's dissect the individual lines to check what they should do:
- _238_: `integrationRuntimeCustomDescription: integrationRuntime.?integrationRuntimeCustomDescription`
  - IF the `integrationRuntimeCustomDescription` is provided, it will use it. Otherwise it will pass `null` which should cause Bicep to automatically pull the parameter's default `'Managed Integration Runtime created by avm-res-datafactory-factories'` from the referenced child-module, which is the same value as before the change   
- _239_: `managedVirtualNetworkName: integrationRuntime.?managedVirtualNetworkName`
  - IF the `managedVirtualNetworkName` is provided, it will use it. Otherwise it will pass `null` which should cause Bicep to automatically pull the parameter's default `''` from the referenced child-module, which is the same value as before the change   
- _240_: `integrationRuntime.?typeProperties`
  - IF the `typeProperties` is provided, it will use it. Otherwise it will pass `null` which should cause Bicep to automatically pull the parameter's default `{}` from the referenced child-module, which is the same value as before the change   

So the changes in these lines don't change anything really, but just upgrade the code to Bicep's capabilities. It's also further ppossible to deploy multiple integration runtimes etc. as the loop is encapsulating the entire `module` block which was not changed. The main change in this module are only the introduction of user-defined-types which provide intellisense to the module's user.